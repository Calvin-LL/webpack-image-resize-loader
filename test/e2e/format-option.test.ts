import { toMatchImageSnapshot } from "jest-image-snapshot";

import WIRLWebpackTestCompiler from "./helpers/WIRLWebpackTestCompiler";

expect.extend({ toMatchImageSnapshot });

describe.each([4, 5] as const)('v%d "format" option', (webpackVersion) => {
  it("should work with jpeg", async () => {
    const compiler = new WIRLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        width: 10,
        format: "jpeg",
      },
      fileLoaderOptions: {
        name: "image.jpg",
      },
    });

    expect(await bundle.readAssetAsPNG("image.jpg")).toMatchImageSnapshot({
      customDiffConfig: { threshold: 0 },
      customSnapshotIdentifier: "10w-80q",
    });
  });

  it("should work with png", async () => {
    const compiler = new WIRLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        width: 10,
        format: "png",
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it("should work with webp", async () => {
    const compiler = new WIRLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        width: 10,
        format: "webp",
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it("should work with tiff", async () => {
    const compiler = new WIRLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        width: 10,
        format: "tiff",
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });
});
