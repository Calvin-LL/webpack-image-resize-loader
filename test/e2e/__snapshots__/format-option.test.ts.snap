// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 "format" option should work with png: result 1`] = `"29ad4d8877eba5bf0dc43f4613cdb42d.png"`;

exports[`v4 "format" option should work with tiff: result 1`] = `"b1aa4799076e75482c8a50b0a29c92f7.tiff"`;

exports[`v4 "format" option should work with webp: result 1`] = `"090bb994797d1a24747f9e30e52cc659.webp"`;

exports[`v5 "format" option should work with png: result 1`] = `"29ad4d8877eba5bf0dc43f4613cdb42d.png"`;

exports[`v5 "format" option should work with tiff: result 1`] = `"b1aa4799076e75482c8a50b0a29c92f7.tiff"`;

exports[`v5 "format" option should work with webp: result 1`] = `"090bb994797d1a24747f9e30e52cc659.webp"`;
